crypto/src/hash.rs

```diff
@@ -556,41 +549,41 @@ pk_with_hash!(PublicKeyP256, ContractTz3Hash);
 pk_with_hash!(PublicKeyBls, ContractTz4Hash);
 
 impl TryFrom<PublicKeyEd25519> for ContractTz1Hash {
-    type Error = TryFromPKError;
+    type Error = ::std::convert::Infallible;
 
     fn try_from(source: PublicKeyEd25519) -> Result<Self, Self::Error> {
         let hash = blake2b::digest_160(&source.0);
-        let typed_hash = Self::from_bytes(&hash)?;
+        let typed_hash = Self::from_bytes(&hash).unwrap();
         Ok(typed_hash)
     }
 }
 
 impl TryFrom<PublicKeySecp256k1> for ContractTz2Hash {
-    type Error = TryFromPKError;
+    type Error = ::std::convert::Infallible;
 
     fn try_from(source: PublicKeySecp256k1) -> Result<Self, Self::Error> {
         let hash = blake2b::digest_160(&source.0);
-        let typed_hash = Self::from_bytes(&hash)?;
+        let typed_hash = Self::from_bytes(&hash).unwrap();
         Ok(typed_hash)
     }
 }
 
 impl TryFrom<PublicKeyP256> for ContractTz3Hash {
-    type Error = TryFromPKError;
+    type Error = ::std::convert::Infallible;
 
     fn try_from(source: PublicKeyP256) -> Result<Self, Self::Error> {
         let hash = blake2b::digest_160(&source.0);
-        let typed_hash = Self::from_bytes(&hash)?;
+        let typed_hash = Self::from_bytes(&hash).unwrap();
         Ok(typed_hash)
     }
 }
 
 impl TryFrom<PublicKeyBls> for ContractTz4Hash {
-    type Error = TryFromPKError;
+    type Error = ::std::convert::Infallible;
```

Comment:
we could remove the `Result` type altogether from `PublicKeyWithHash` no?

Reply:
I was a bit concerned about changing the trait interface wrt downstream compatibility, but arguably this isn't that big of a concern. Done in f105e32d21ef6ae79432cbe6299b983ea75377a3

Reply:
thanks! will be a few things needing to be updated downstream anyway - easier to do it all in one go 😄